### ISSUE TITLE
Compilation warnings on mac

### ISSUE THREAD

**Issue by seisman** [GenericMappingTools/gmt#843]
**Description of the problem**

Warning message:
```
../src/gmt_plot.c:5811:36: warning: format string is not a string literal (potentially insecure) [-Wformat-security]
                        snprintf (scale_c, GMT_LEN32-1, pch);
                                                        ^~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/usr/include/secure/_stdio.h:57:62: note: expanded from macro 'snprintf'
  __builtin___snprintf_chk (str, len, 0, __darwin_obsz(str), __VA_ARGS__)
                                                             ^~~~~~~~~~~
../src/gmt_plot.c:5811:36: note: treat the string as an argument to avoid this
                        snprintf (scale_c, GMT_LEN32-1, pch);
                                                        ^
                                                        "%s",
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/usr/include/secure/_stdio.h:57:62: note: expanded from macro 'snprintf'
  __builtin___snprintf_chk (str, len, 0, __darwin_obsz(str), __VA_ARGS__)
                                                             ^
../src/gmt_plot.c:5831:7: warning: address of array 'scale_c' will always evaluate to 'true' [-Wpointer-bool-conversion]
                if (scale_c) strcat (szProj4, scale_c);         /* Add the width/scale found above */
                ~~  ^~~~~~~
2 warnings generated.
```

**Comment by PaulWessel** [comment]
Should be if (scale_c[0]) presumably?
Just use scale_c, GMTLEN32-1, "%s", pch as args

**Comment by joa-quim** [comment]
Fixed in #842
